### PR TITLE
Extract Prefresh code into a separate plugin

### PR DESCRIPTION
**Pull request by marvinhagemeister**

### CODE REVIEW

**Review by developit — APPROVED** [review]
LGTM!

File: packages/wmr/src/plugins/preact/prefresh.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import MagicString from 'magic-string';
+import path from 'path';
+
+const PREFRESH = `
+if (import.meta.hot) {
```
Comment (developit) [inline packages/wmr/src/plugins/preact/prefresh.js:5]:
```suggestion
import '@prefresh/core';
if (import.meta.hot) {
```

File: packages/wmr/src/plugins/preact/prefresh.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			// Only inject into modules with JSX and exports
+			if (!hasJsx || !hasExport) return;
+
+			const s = new MagicString(code, {
+				filename: id,
+				// @ts-ignore
+				indentExclusionRanges: undefined
+			});
+
+			s.prepend(`import '@prefresh/core';\n`);
```
Comment (developit) [inline packages/wmr/src/plugins/preact/prefresh.js:46]:
```suggestion
```

**Comment by developit** [comment]
The CI failure is interesting... it's not getting HMR injected for some reason, but I can't see how this PR would cause that to happen.

**Comment by marvinhagemeister** [comment]
@developit The CI failure is caused by a commit of mine that was intended for another branch 😬 Dropping that makes it pass again.

**Pull request merged**